xalia/Win32/UiaProvider.cs

```diff
@@ -630,10 +630,17 @@ private ElementIdentifier ElementIdFromFragmentBackground(IRawElementProviderFra
 
             if (hwnd == 0)
             {
-                var value = simple.GetPropertyValue(UIA_NativeWindowHandlePropertyId);
-                if (!(value is null))
+                try
                 {
-                    hwnd = (int)value;
+                    var value = simple.GetPropertyValue(UIA_NativeWindowHandlePropertyId);
+                    if (!(value is null))
+                    {
+                        hwnd = (int)value;
+                    }
+                }
+                catch (NullReferenceException)
+                {
+                    // Not sure why this can happen
                 }

```

Comment:
Does Mono COM interop automatically set `VARIANT`s to VT_EMPTY? It's possible that if this particular COM interface doesn't handle  `UIA_NativeWindowHandlePropertyId` (which technically, only the clientside HWND provider and the HostRawElement provider are supposed to handle) and it ends up uninitialized.

Reply:
This was on Windows. I'm pretty sure it'll initialize `out` parameters, but if it doesn't then COM RPC should, otherwise it might try to marshal invalid data.

I guess it's possible the method returned E_POINTER.